src/meta/src/stream/stream_manager.rs

```diff
@@ -368,13 +392,14 @@ where
             // Create empty locations.
             let mut locations = ScheduledLocations::with_workers(workers);
 
-            // Schedule each fragment(actors) to nodes, recorded in `locations`.
+            // Schedule each fragment(actors) to nodes except chain, recorded in `locations`.
             // Vnode mapping in fragment will be filled in as well.
             let topological_order = table_fragments.generate_topological_order();
             for fragment_id in topological_order {
                 let fragment = table_fragments.fragments.get_mut(&fragment_id).unwrap();
-                self.scheduler.schedule(fragment, &mut locations).await?;
-                table_fragments.update_table_fragment_map(fragment_id);
+                if !fragment_upstream_table_map.contains_key(&fragment_id) {
```

Comment:
Ditto. I think it'll be clearer if we decide by `FragmentType`. 🥵

Reply:
I wish it could be decided by `FragmentType`. The only way is to make fragment have multiple types, but this will affect a lot in some of functions.